### PR TITLE
Added alert_email field

### LINKED ISSUE #8: alert_email field missing from API calls
The dead man's snitch API has an alert_email field that needs to be added to this terraform module. See https://deadmanssnitch.com/docs/api/v1#creating-a-snitch

### PR DESCRIPTION
**Pull request by richarddouglasfrye**

### CODE REVIEW

File: dmsnitch/resource_snitch.go [hunk 1]
```diff
@@ -82,6 +91,12 @@ func resourceSnitch() *schema.Resource {
 }
 
 func newSnitchFromResource(d *schema.ResourceData) *Snitch {
+	email := make([]string, 0, len(d.Get("alert_email").(*schema.Set).List()))
```
Comment (plukevdh) [inline dmsnitch/resource_snitch.go:94]:
```suggestion
	emails := make([]string, 0, len(d.Get("alert_emails").(*schema.Set).List()))
```
I'd prefer this var be named plural to reflect the fact that it is an array

File: dmsnitch/resource_snitch.go [hunk 2]
```diff
@@ -82,6 +91,12 @@ func resourceSnitch() *schema.Resource {
...
+
+	for _, item := range d.Get("alert_email").(*schema.Set).List() {
+		email = append(email, item.(string))
```
Comment (plukevdh) [inline dmsnitch/resource_snitch.go:97]:
```suggestion
		emails = append(emails, item.(string))
```
Same change here

File: dmsnitch/resource_snitch.go [hunk 3]
```diff
@@ -95,6 +110,7 @@ func newSnitchFromResource(d *schema.ResourceData) *Snitch {
 		Status:   d.Get("status").(string),
 		Interval: d.Get("interval").(string),
 		Type:     d.Get("type").(string),
+		Email:    email,
```
Comment (plukevdh) [inline dmsnitch/resource_snitch.go:113]:
```suggestion
		AlertEmails:    emails,
```
Same change again here

File: dmsnitch/resource_snitch.go [hunk 4]
```diff
@@ -16,6 +17,7 @@ type Snitch struct {
 	Status   string   `json:"status,omitempty"`
 	Interval string   `json:"interval,omitempty"`
 	Type     string   `json:"alert_type,omitempty"`
+	Email    []string `json:"alert_email,omitempty"`
```
Comment (plukevdh) [inline dmsnitch/resource_snitch.go:20]:
```suggestion
	AlertEmails   []string `json:"alert_email,omitempty"`
```
Let's be specific what "email" is for

File: README.md [hunk 5]
```diff
@@ -50,6 +50,7 @@ resource "dmsnitch_snitch" "mysnitch" {
   
   interval = "daily" 
   type = "basic"
+  email = ["example@example.com", "support@example.com"]
```
Comment (plukevdh) [inline README.md]:
```suggestion
  alert_emails = ["example@example.com", "support@example.com"]
```

File: README.md [hunk 6]
```diff
@@ -96,6 +97,7 @@ You can also import existing snitches using their token found in the snitch's pa
 | `notes`| no | | `Managed by Terraform` | 
 | `interval` | yes | `15_minute`, `30_minute`, `hourly`, `daily`, `weekly`, `monthly` | `daily` |
 | `type` | yes; `smart` is only valid for `weekly` or `monthly` intervals  | `basic`, `smart` | `basic` |
+| `email` | no | an array of values |
```
Comment (plukevdh) [inline README.md]:
```suggestion
| `alert_emails` | no | an array of values |  
```

File: dmsnitch/resource_snitch.go [hunk 7]
```diff
@@ -60,6 +62,13 @@ func resourceSnitch() *schema.Resource {
 				Default:  "basic",
 			},
 
+			"email": {
```
Comment (plukevdh) [inline dmsnitch/resource_snitch.go]:
```suggestion
			"alert_emails": {
```

File: dmsnitch/resource_snitch.go [hunk 8]
```diff
@@ -154,6 +170,12 @@ func resourceSnitchRead(d *schema.ResourceData, m interface{}) error {
 			return decodeerr
 		}
 
+		emailList := make([]string, 0, len(snitch.Email))
```
Comment (plukevdh) [inline dmsnitch/resource_snitch.go:173]:
```suggestion
		emailList := make([]string, 0, len(snitch.AlertEmails))
```

File: dmsnitch/resource_snitch.go [hunk 9]
```diff
@@ -154,6 +170,12 @@ func resourceSnitchRead(d *schema.ResourceData, m interface{}) error {
...
+
+		for _, event := range snitch.Email {
```
Comment (plukevdh) [inline dmsnitch/resource_snitch.go:175]:
```suggestion
		for _, event := range snitch.AlertEmails {
```

File: dmsnitch/resource_snitch.go [hunk 10]
```diff
@@ -166,6 +188,7 @@ func resourceSnitchRead(d *schema.ResourceData, m interface{}) error {
 		d.Set("status", snitch.Status)
 		d.Set("interval", snitch.Interval)
 		d.Set("type", snitch.Type)
+		d.Set("tags", emailList)
```
Comment (plukevdh) [inline dmsnitch/resource_snitch.go:191]:
```suggestion
		d.Set("alert_emails", emailList)
```
This looks like a copy/paste error

File: dmsnitch/resource_snitch.go [hunk 11]
```diff
@@ -82,6 +91,12 @@ func resourceSnitch() *schema.Resource {
 }
 
 func newSnitchFromResource(d *schema.ResourceData) *Snitch {
+	email := make([]string, 0, len(d.Get("alert_email").(*schema.Set).List()))
+
+	for _, item := range d.Get("alert_email").(*schema.Set).List() {
```
Comment (plukevdh) [inline dmsnitch/resource_snitch.go:96]:
```suggestion
	for _, item := range d.Get("alert_emails").(*schema.Set).List() {
```

**Review by plukevdh — CHANGES_REQUESTED** [review]
I'm proposing a number of variable and property name changes simply to reflect the purpose of the "email" field and to denote the fact that it is an array (much like "tags").

File: dmsnitch/resource_snitch.go (hunk 1 above)
Comment (richarddouglasfrye) [reply dmsnitch/resource_snitch.go:94]: I agree but the field in the json from their API is alert_email. That is why I kept the naming consistent with how they named it.  Do you want a change to plural everywhere except the actual field or leave it how I originally had it?

**Comment by richarddouglasfrye** [comment]
The field in their API is called alert_email. They did not make it plural but I agree that they should have. So we can make it plural everywhere in the provider or match their API. I prefer to match the API to keep down confusion. Please let me know how you want to proceed.

**Comment by plukevdh** [comment]
@richarddouglasfrye Sorry for the delay in responding. My personal design preference for the APIs _Terraform_ provides is to use language to indicate intent, even if the underlying _API_ is inconsistent. Terraform is providing a layer of abstraction on top of a large number of third-party APIs in order to improve visibility and clarity of intent (infra-as-code). I don't feel a need to map directly to the underlying API naming conventions, particularly if I think they are badly named 😄

File: README.md [hunk 12]
```diff
@@ -1,26 +1,13 @@
...
+- Terraform 0.13.x or higher
 - Go 1.11 or higher
 - Go module support
 
-Please note [the following details](https://www.terraform.io/docs/extend/terraform-0.12-compatibility.html) if you have built this plugin prior to Terraform 0.12. 
-
 ## Setting up the provider
 
 ```sh
-$ go get -u github.com/plukevdh/terraform-provider-dmsnitch
```
Comment (plukevdh) [inline README.md:23]: These changes will not be accepted if you want them merged upstream for the registry.

File: main.go [hunk 13]
```diff
@@ -2,7 +2,7 @@ package main
 
 import (
 	"github.com/hashicorp/terraform/plugin"
-	"github.com/plukevdh/terraform-provider-dmsnitch/dmsnitch"
```
Comment (plukevdh) [inline main.go:5]: this needs to be reverted before merge

File: go.mod [hunk 14]
```diff
@@ -1,4 +1,4 @@
-module github.com/plukevdh/terraform-provider-dmsnitch
```
Comment (plukevdh) [inline go.mod:1]: this needs to be reverted before merge

**Comment by pomegranited** [comment]
Hi! We need this feature in order to use your excellent repo, so was really glad to see this change in progress.

@richarddouglasfrye do you know when you'll have a chance to address @plukevdh 's comments?
Or are you ok with me creating a new PR starting with your changes and addressing them myself?

**Comment by yurrriq** [comment]
#13 was merged, where I tried to address the feedback. Thanks again to @richarddouglasfrye and @plukevdh!

**Comment by plukevdh** [comment]
Closed in favor of #13

**Pull request closed without merging**